src/mqt/bench/benchmark_generator.py

```diff
@@ -389,37 +398,39 @@ def get_benchmark(
         if compiler == "tket":
             return tket_helper.get_indep_level(qc, circuit_size, False, True)
 
-    if gate_set_name not in utils.get_supported_gatesets():
-        msg = f"Selected gate_set_name must be in {utils.get_supported_gatesets()}."
+    if provider_name not in get_available_provider_names():
+        msg = f"Selected provider_name must be in {get_available_provider_names()}."
         raise ValueError(msg)
 
     native_gates_level = 2
     if level in ("nativegates", native_gates_level):
-        assert gate_set_name is not None
+        assert provider_name is not None
+        provider = get_provider_by_name(provider_name)
         if compiler == "qiskit":
             assert compiler_settings.qiskit is not None
             opt_level = compiler_settings.qiskit.optimization_level
-            return qiskit_helper.get_native_gates_level(qc, gate_set_name, circuit_size, opt_level, False, True)
+            return qiskit_helper.get_native_gates_level(qc, provider, circuit_size, opt_level, False, True)
         if compiler == "tket":
-            return tket_helper.get_native_gates_level(qc, gate_set_name, circuit_size, False, True)
+            return tket_helper.get_native_gates_level(qc, provider, circuit_size, False, True)
 
-    if device_name not in utils.get_supported_devices():
-        msg = f"Selected device_name must be in {utils.get_supported_devices()}."
+    if device_name not in devices.get_available_device_names():
+        msg = f"Selected device_name must be in {devices.get_available_device_names()}."
         raise ValueError(msg)
 
     mapped_level = 3
     if level in ("mapped", mapped_level):
-        assert gate_set_name is not None
+        assert provider_name is not None
+        provider = get_provider_by_name(provider_name)
```

Comment:
Would it be possible to remove the usage of the parameter `provider_name` for the mapped level benchmarks? The rationale behind is that, coming from a user's perspective, it feels redundant to provide both `provider_name` and `device_name`, since the former could be inferred from the latter I assume.